### ISSUE TITLE
Memory Leak for a while of application running

### ISSUE THREAD

**Issue by zjunweihit** [RadeonOpenCompute/ROCK-Kernel-Driver#84]
**[Environment]**
* Ubuntu 18.04.3
* kernel 4.15.0-70-generic
* rocm-2.9
* Vega20 VII

**[Issue]**
* Generally running an application for a long time, we can see memory leak
* A simple way is to run kfd test and kill it. (BTW, in my PC, I cannot finish this test, it will be killed when the test running, got the same results)
```
./kfdtest --gtest_filter="KFDMemoryTest.*"
<Ctrl-c>
```
* the VRAM is not freed, even after the test exits.
```
Total(LOC -  6.48%):    16368MB | Used:     1061MB | Free:    15307MB |        
```
* the ttm bo workqueue is till here, trying to remove the bo
```
     kworker/8:1-30524 [008] .... 520814.097399: workqueue_execute_start: work struct 000000006619786f: function ttm_bo_delayed_workqueue [amdttm]
          <idle>-0     [008] d.s. 520814.133388: workqueue_queue_work: work struct=000000006619786f function=ttm_bo_delayed_workqueue [amdttm] workqueue=00000000b0b35bb9 req_cpu=8192 cpu=8
     kworker/8:1-30524 [008] .... 520814.133407: workqueue_execute_start: work struct 000000006619786f: function ttm_bo_delayed_workqueue [amdttm]
```

**[additionally]**
* it works fine on rocm-2.2
* Guess it's about the amdgpu driver during kernel transition, from 4.18.0 to 5.0.0-rc1 on dkms support.

**Comment by zjunweihit** [comment]
@kentrussell @fxkamd could you help to check it?

**Comment by kentrussell** [comment]
@fxkamd I believe you have a fix for this slated for 3.2?

**Comment by fxkamd** [comment]
I submitted a fix for a memory leak after process termination yesterday.

**Comment by kentrussell** [comment]
Should be resolved in 3.3
